mathics/builtin/drawing/graphics3d.py

```diff
@@ -243,29 +250,47 @@ class Cuboid(Builtin):
 
     summary_text = "unit cube"
 
-    def apply_unit_cube(self, xmin, ymin, zmin, evaluation):
-        "Cuboid[{xmin_, ymin_, zmin_}]"
+    def apply_rectangle(self, xmin, ymin, xmax, ymax, evaluation):
+        "Cuboid[{xmin_, ymin_}, {xmax_, ymax_}]"
 
-        return Expression(
-            "Cuboid",
-            List(
-                List(xmin, ymin, zmin),
-                List(
-                    Real(xmin.to_python() + 1),
-                    Real(ymin.to_python() + 1),
-                    Real(zmin.to_python() + 1),
-                ),
-            ),
-        )
+        return Expression("Rectangle", List(xmin, ymin), List(xmax, ymax))
 
-    def apply_check(self, positions, evaluation):
+    def apply(self, positions, evaluation):
         "Cuboid[positions_List]"
 
-        if len(positions.get_leaves()) % 2 == 1:
+        if not isinstance(positions.get_leaves()[0], list):
+            positions = List(positions)
+
+        coordinates = Coordinates(positions)
+
+        if len(coordinates) == 1:
+            if coordinates.dimension == 2:
+                return Expression(
+                    "Rectangle",
+                    List(*(coordinates.coordinates[0])),
+                    List(
+                        Real(coordinates.coordinates[0][0] + 1),
+                        Real(coordinates.coordinates[0][1] + 1),
+                    ),
+                )
+
+            return Expression(
+                "Cuboid",
+                List(
+                    List(*(coordinates.coordinates[0])),
+                    List(
+                        Real(coordinates.coordinates[0][0] + 1),
```

Comment:
Personally I would prefer `.x`, `.y` and `.z` properties on the coordinates. Using the properties disallows setting them, and it is clearer. 

I see this kind of thing happen a lot in mathics-threejs-backend and thought the same thing.